openlibrary/plugins/upstream/utils.py

```diff
@@ -854,6 +855,50 @@ def today():
     return datetime.datetime.today()
 
 
+class HTMLTagRemover(HTMLParser):
+
+    def __init__(self):
+        super().__init__()
+        self.data = []
+
+    def handle_data(self, data):
+        self.data.append(data)
+
+
+@public
+def reformat_html(html_str, max_length=None):
+    """
+    Reformats an HTML string, removing all opening and closing tags.
+    Adds a line break element between each set of text content.
+    Optionally truncates contents that exceeds the given max length.
+
+    returns: A reformatted HTML string
+    """
+    parser = HTMLTagRemover()
+    parser.feed(html_str)
+
+    content = [s.strip() for s in parser.data if len(s.strip())]
+
+    if max_length:
+        str_lengths = [len(s) for s in content]
+        end_index = 0
+        for i in str_lengths:
+            end_index += 1
+            if max_length - i == 0:
+                break
+            elif max_length - i < 0:
+                content[end_index - 1] = truncate(
+                    content[end_index - 1],
+                    max_length
+                )
+                break
+            max_length -= i
+    else:
+        end_index = len(content) + 1
+
+    return '<br>'.join(content[:end_index])
```

Comment:
We might be able to get rid of that big `if` block if we do something like this:

```py
truncate('\n'.join(content), max_length).replace('\n', '<br>')
```